where_art_though.js

```diff
@@ -0,0 +1,40 @@
+/* 
+Make a function that looks through an array of objects (first argument) and returns an array of all objects that have 
+matching property and value pairs (second argument). Each property and value pair of the source object has to be present
+in the object from the collection if it is to be included in the returned array.
+
+For example, if the first argument is 
+[{ first: "Romeo", last: "Montague" }, { first: "Mercutio", last: null }, { first: "Tybalt", last: "Capulet" }], 
+and the second argument is { last: "Capulet" }, then you must return the third object from the array (the first argument), 
+because it contains the property and it's value, that was passed on as the second argument.
+*/
+
+function where(collection, source) {
+  
+  var arr = [];
+  var sourceKeys = Object.keys(source);
+  // delcaration out of for loop to save memory
+  var check;
+  
+  // for each object in collection
+  for(var i = 0; i < collection.length ; i++){
+    // for each key in source object
```

Comment:
If source is empty, ie everything should pass, you will access an un-initialized variable. The default value will be `undefined` I believe which is falsish and you will not add to the collection.
`var check = true;`